Framework/src/TaskRunner.cxx

```diff
@@ -182,11 +181,37 @@ void TaskRunner::populateConfig(std::string taskName)
     mTaskConfig.cycleDurationSeconds = taskConfigTree->second.get<int>("cycleDurationSeconds", 10);
     mTaskConfig.maxNumberCycles = taskConfigTree->second.get<int>("maxNumberCycles", -1);
 
-    std::string policiesFilePath = mConfigFile->get<std::string>("dataSamplingPolicyFile", "");
-    if (policiesFilePath.empty()) {
-      mInputSpecs = framework::DataSampling::InputSpecsForPolicy(mConfigFile.get(), taskConfigTree->second.get<std::string>("dataSamplingPolicy"));
+    auto policiesFilePath = mConfigFile->get<std::string>("dataSamplingPolicyFile", "");
+    ConfigurationInterface* config = policiesFilePath.empty() ? mConfigFile.get() : ConfigurationFactory::getConfiguration(policiesFilePath).get();
+    auto policiesTree = config->getRecursive("dataSamplingPolicies");
+    auto dataSourceTree = taskConfigTree->second.get_child("dataSource");
+    std::string type = dataSourceTree.get<std::string>("type");
+
+    if (type == "dataSamplingPolicy") {
+      auto policyName = dataSourceTree.get<std::string>("name");
+      LOG(INFO) << "policyName : " << policyName;
+      mInputSpecs = framework::DataSampling::InputSpecsForPolicy(config, policyName);
+    } else if (type == "direct") {
+
+      auto subSpecString = dataSourceTree.get<std::string>("subSpec");
+      LOG(INFO) << "subSpecString : " << subSpecString;
```

Comment:
I guess this specific log is a leftover from the development. If you want it, the full InputSpec has the operator<< overloaded.

Reply:
Totally, I will remove it now.